rt/features/metrics/src/main/java/org/apache/cxf/metrics/ExchangeMetrics.java

```diff
@@ -40,9 +40,11 @@ public ExchangeMetrics(Exchange e) {
     }
 
     public ExchangeMetrics addContext(MetricsContext ctx) {
-        contexts.addLast(ctx);
-        if (started) {
-            ctx.start(exchange);
+        if (!contexts.contains(ctx)) {
```

Comment:
I think the question I have - why do we have context registered more than once? I suspect it may have impact in other places as well

Reply:
Thank you @ffang